lib/SyTest/Homeserver.pm

```diff
@@ -59,19 +63,101 @@ sub write_json_file
    return $self->write_file( $relpath, JSON::encode_json( $content ) );
 }
 
-sub clear_db_sqlite
+=head2 _get_dbconfig
+
+   %db_config = $self->_get_dbconfig( %defaults )
+
+This method loads the database config from C<database.yaml>, or creates that
+file according to the given defaults.
+
+It then passes the loaded config to C<_check_db_config> for
+sanity-checking. That method may be overridden by subclasses, and should C<die>
+if there is a problem with the config.
+
+Finally, it calls the relevant clear_db method to clear out the configured
+database.
+
+It returns the config hash.
+
+=cut
+
+sub _get_dbconfig
+{
+   my $self = shift;
+   my ( %defaults ) = @_;
+
+   my $hs_dir = $self->{hs_dir};
+   my $db_config_path = "database.yaml";
+   my $db_config_abs_path = "$hs_dir/${db_config_path}";
+
+   my ( %db_config );
+   if( -f $db_config_abs_path ) {
+      %db_config = %{ YAML::LoadFile( $db_config_abs_path ) };
+
+      # backwards-compatibility hacks
+      my $db_name = delete $db_config{name};
+      if( defined $db_name ) {
+         if( $db_name eq 'psycopg2' ) {
+            $db_config{type} = 'pg';
+         } elsif( $db_name eq 'sqlite3' ) {
```

Comment:
Small style - uncuddle the `elsif` onto its own line

    }
    elsif( COND ) {

Reply:
fixed in f9c4876